vendor/github.com/ethereum/go-ethereum/whisper/whisperv5/whisper.go

```diff
@@ -540,16 +542,48 @@ func (w *Whisper) Stop() error {
 	return nil
 }
 
+func waitTrusted(peer *Peer) error {
+	timeout := time.After(time.Minute)
+	ticker := time.NewTicker(2 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-timeout:
+			return p2p.DiscTooManyPeers
```

Comment:
How do we know that there are too many peers at this point? Maybe some got disconnected meanwhile and we should not disconnect this one.

Reply:
p2p.Server will re-dial in this case, but right it won't hurt to check if another peer was disconnected